tests/unit/test_run.py

```diff
@@ -335,6 +335,23 @@ def test_init_self(test_iam_df, data_cls):
     assert_scmdf_almost_equal(a, b)
 
 
+def test_init_with_metadata(test_scm_run):
+    expected_metadata = {"test": "example"}
+    b = ScmRun(test_scm_run.timeseries(), metadata=expected_metadata)
+
+    assert b.metadata == expected_metadata
```

Comment:
add a check that a copy is made i.e. `assert id(expected_metadata) != id(b.metadata)`? I think that's the expected behaviour